src/libraries/System.Private.CoreLib/src/System/SpanHelpers.ByteMemOps.cs

```diff
@@ -498,24 +506,16 @@ internal static void Fill(ref byte dest, byte value, nuint len)
                 nuint stopLoopAtOffset = len & ~(nuint)7;
                 do
                 {
-                    Unsafe.Add(ref dest, (nint)i + 0) = value;
-                    Unsafe.Add(ref dest, (nint)i + 1) = value;
-                    Unsafe.Add(ref dest, (nint)i + 2) = value;
-                    Unsafe.Add(ref dest, (nint)i + 3) = value;
-                    Unsafe.Add(ref dest, (nint)i + 4) = value;
-                    Unsafe.Add(ref dest, (nint)i + 5) = value;
-                    Unsafe.Add(ref dest, (nint)i + 6) = value;
-                    Unsafe.Add(ref dest, (nint)i + 7) = value;
+                    // broadcast the value to all 8 bytes of the ulong and write it to memory
+                    Unsafe.WriteUnaligned(ref Unsafe.Add(ref dest, i), value * 0x101010101010101ul);
```

Comment:
it's not just for Mono, but it's shorter (and faster)

Reply:
This is 64-bit multiplication. Is it faster on 32-bit platforms as well?